src/com/model/BanknotesCounter.java

```diff
@@ -0,0 +1,31 @@
+package com.model;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class BanknotesCounter {
+    // Allowed banknotes to be deposited;
+    public final int[] ALLOWED_BANKNOTES = {1, 3, 5, 10, 20, 100};
+    // The key is the denomination of the banknote. Value - its quantity in stock;
+    public Map<Integer, Integer> banknotesMap = new HashMap<>();
+    public int totalSum;
+
+    public BanknotesCounter() {
+        totalSum = 0;
+        for (int allowedBanknote : ALLOWED_BANKNOTES) {
+            banknotesMap.put(allowedBanknote, 0);
+        }
+    }
+
+    public void incrementBanknoteCounter(int banknote) {
+        Integer banknoteAmount = banknotesMap.get(banknote);
```

Comment:
Prefer "unboxed" primitive values (i.e., `int`) when working with primitives. It doesn't matter that the map is parametrised with `Integer`

Reply:
(cosmetic) You could also use the "atomic" map operation like `banknotesMap.computeIfPresent()`  instead of reading value and then changing it. This is better for multithreaded implementations, however in this particular case doesn't really matter